### PR TITLE
Changing the dataset to produce train/val/test splits

### LINKED ISSUE #46: Discussion: Train/val/test splitting and dataset generation
As mentioned in #12, we want to split the data in the different partitions, and need to decide how to.

### PR DESCRIPTION
**Pull request by hzavadil98**
As discussed yesterday I propose to change the datasets so that they returned train validation and test(evaluation) splits as is usual. I adjusted the train validation and test loops in main accordingly. This will require a change in your datasets - now the `split = ["train","validation","test"]` is passed to it along with `split_percentage=(0,1)` which indicates the percentage of observations from the training part of data to be used as validation (defaults to 0.2). I am dealing with it by using fixed seed in  `torch.utils.data.random_split` to split the training data if `split = "training"or"validation"`, feel free to copy.  Closes #46 relates to #49 . If everybody on board please push your adjusted datasets into this draft PR.

### CODE REVIEW

**Comment by salomaestro** [comment]
@hzavadil98 Maybe I could propose a bit of a different way to split the data? I'll try to show what I'm thinking here, then perhaps create a PR to merge with this branch if we like it?

So if we structure the downloading of the data differently, I think we could simplify the setup a bit:

1. Move the downloading to a separate function/class, e.g.:
```python
class Downloader:
    def mnist(to_dir):
        ...
        return path_to_mnist_dir_or_file, number_of_samples

    def svhn(to_dir):
        ...
        return ...

    def usps(to_dir):
        ...
        return ...
```
2. In `main.py` the dataset generation would be:
```python
data_path, n_samples = Downloader.mnist(data_path)

train_samples, val_samples = torch.utils.data.random_split(range(n_samples), [1 - val_size, val_size])

train_dataset = load_data(data_path, train_samples)
val_dataset = load_data(data_path, val_samples)
```

Any thoughts?

**Comment by hzavadil98** [comment]
Closing this pull request since it was already merged into #54
